### ISSUE TITLE
Truthy initial loading state when using manual

### ISSUE THREAD

**Issue by Wgil** [simoneb/axios-hooks#284]
I'm using the `manual` option like this:

```tsx
const [response, execute] = useAxios<APIResponse<T> | undefined>(
    {
      url: `${REACT_APP_API_URL}/${endpoint}`,
      method: data ? 'POST' : 'GET'
    },
    { manual: true, useCache: false }
  )

  useEffect(() => {
    const fetch = async () => {
      const user = firebase.auth().currentUser
      if (!user) return

      const token = await user.getIdToken(/* forceRefresh */ true)
      if (!token) return

      execute({
        headers: {
          authorization: `Bearer ${token}`
        },
        data
      })
    }

    fetch()
  }, [execute, data])
```

My issue is that since it's a manual request, the loading state is `false` until `execute` gets called. Is there any way I can add initialize loading state as `true` using manual?

**Comment by simoneb** [comment]
No, there is no way that loading can be true if the request is not executing. I do see why you want that though, and I think the solution is to structure your application in a different way. I see two options here:

- you handle the axios configuration at the root level of the application and only once you've obtained the user token you initialize axios-hooks using `configure` then start your application
- you do a more localized solution, similar to the above. you can extract the token-obtaining logic in a hook (see pseudo code below) and conditionally execute the axios-hooks request based on whether the user is there or not

```js
// pseudocode

function useToken() {
  const [token, setToken] = useState()

  useEffect(() => {
    const fetch = async () => {
      const user = firebase.auth().currentUser
      if (!user) return

      const token = await user.getIdToken(/* forceRefresh */ true)
      if (!token) return

      setToken(token)
    }

    fetch()
  }, [])

  return token
}

function YourComponent() {
  const token = useToken()

  const [response] = useAxios({
      url: `${REACT_APP_API_URL}/${endpoint}`,
      method: data ? 'POST' : 'GET'
      headers: { authorization: `Bearer ${token}`
    },
    { manual: !token, useCache: false }
  )
}

**Comment by Wgil** [comment]
I went for option two but added a custom hook since I want my loading state to be truthy while I'm fetching the token:

```ts
export const useCustomAxios = <T>(endpoint: string, data?: any) => {
  const token = useToken()
  const [response, refetch] = useAxios<APIResponse<T> | undefined>(
    {
      headers: {
        authorization: `Bearer ${token}`
      },
      url: `${REACT_APP_API_URL}/${endpoint}`,
      method: data ? 'POST' : 'GET',
      data
    },
    { manual: !token, useCache: false }
  )

  return [{ ...response, loading: !token && response.loading }, refetch] as [
    ResponseValues<APIResponse<T> | undefined>,
    (
      config?: AxiosRequestConfig,
      options?: RefetchOptions
    ) => AxiosPromise<APIResponse<T> | undefined>
  ]
}
```

On another note, I think this type:
```ts
export interface ResponseValues<T> {
  data: T
  loading: boolean
  error?: AxiosError
  response?: AxiosResponse
}
```
should be
```ts
export interface ResponseValues<T> {
  data: T | undefined
  loading: boolean
  error?: AxiosError
  response?: AxiosResponse
}
```
since `data` is not there until we get a response. Should I open another issue for this?
Thanks for helping.

**Comment by simoneb** [comment]
> since `data` is not there until we get a response. Should I open another issue for this?

Thanks, I pushed a fix for this already. Good catch.
